demos/notebooks/demo_pipeline.ipynb

```diff
@@ -212,11 +302,26 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "#%% start a cluster for parallel processing (if a cluster already exists it will be closed and a new session will be opened)\n",
-    "if 'dview' in locals():\n",
+    "print(f\"You have {psutil.cpu_count()} CPUs available in your current environment\")\n",
+    "processors_to_use = None  # Set to number less than cpu_count - 1 if you have memory problems"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#start a cluster for parallel processing \n",
+    "# note if a cluster already exists it will be closed so a new session will be opened\n",
+    "if 'dview' in locals():  # locals contains list of current local variables\n",
```

Comment:
We should review why we're doing this; it should be clear whether dview has a value or not?